src/main/java/com/zaxxer/hikari/metrics/prometheus/HikariCPCollector.java

```diff
@@ -52,12 +57,33 @@ protected HikariCPCollector add(String name, PoolStats poolStats) {
    }
 
    private GaugeMetricFamily createGauge(String metric, String help,
-      Function<PoolStats, Integer> metricValueFunction) {
+                                         Function<PoolStats, Integer> metricValueFunction) {
       GaugeMetricFamily metricFamily = new GaugeMetricFamily(metric, help, LABEL_NAMES);
       poolStatsMap.forEach((k, v) -> metricFamily.addMetric(
          Collections.singletonList(k),
          metricValueFunction.apply(v)
       ));
       return metricFamily;
    }
+
+   @Override
+   public <T extends Collector> T register() {
+      if (registered.contains(CollectorRegistry.defaultRegistry)) {
+         return (T) this;
+      } else {
+         registered.add(CollectorRegistry.defaultRegistry);
+         return super.register();
+      }
```

Comment:
Not thread-safe, despite use of a wrapped ConcurrentHashMap. registered.contains() and registered.add() are not collectively atomic.